test/steps_test.go

```diff
@@ -112,6 +117,16 @@ var contractTestFixture struct {
 	dynamicFee         templates.DynamicFee
 }
 
+var tealCompleResult struct {
+	status   int
```

Comment:
I think that go has error types that you can compare against instead of adding the status to the result here.

The lack of status output in the JS / Python / Go implementations is pretty unfortunate. I don't think there is even a workaround in JS / Python like there is with Go.

Reply:
at the end we end up with the same `if/else`
```
	if err == nil {
		tealCompleResult.status = 200
		tealCompleResult.response = result
		return
	}
	if _, ok := err.(commonV2.BadRequest); ok {
		tealCompleResult.status = 400
		tealCompleResult.response.Hash = ""
		tealCompleResult.response.Result = ""
		return nil
	}
```
but not in `tealCompile` but in `tealCheckCompile`. Do not see a reason to move it there.

Reply:
Sorry, I meant that you don't need to return the status field because it's indirectly returned via the err object.

Reply:
here?
```
result, err := aclv2.TealCompile(tealProgram).Do(context.Background())
```

err will be either `nil` or `commonV2 .BadRequest` and still needs to checked against status code from the test spec.